### PR TITLE
Add support for testing whether a value is a constant-cased string

### LINKED ISSUE #534: [RFC]: add supporting for testing whether a value is a constant-cased string
### Description

This RFC proposes to add support for testing whether a value is a constant-cased string.

```javascript
var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
var constantcase = require( '@stdlib/string/base/constantcase' );

function isConstantcase( value ) {
    return ( isString( value ) && constantcase( value ) === value );
}
```

Package: `@stdlib/assert/is-constantcase`
alias: `isConstantcase`

### Related Issues

No.

### Questions

No.

### Other

Reference packages:

-   [`@stdlib/assert/is-uppercase`](https://github.com/stdlib-js/stdlib/tree/13c1fa39ff73208d819697dc07917bdb7539f295/lib/node_modules/%40stdlib/assert/is-uppercase)

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #534.

## Checklist

- [x] Add readme.md
- [x] Add package.json
- [x] Add lib
- [x] Add examples
- [x] Add docs
- [x] Add etc
- [x] Add bin
- [x] Add benchmark
- [x] Add test

* * *

@kgryte 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Pranavchiku** [comment]
I don't have the access to add assignees for the PR :/

Edit: Thanks :)

**Comment by kgryte** [comment]
@Pranavchiku Sorry about that. This is the first time I'm trying this out. Just assigned. Let's see what happens.

**Comment by kgryte** [comment]
Looks like the workflow is stuck in a queue, so we'll need to wait a while for it to kick off. How long we'll have to wait is TBD.

**Comment by Pranavchiku** [comment]
>Looks like the workflow is stuck in a queue, so we'll need to wait a while for it to kick off. How long we'll have to wait is TBD.

It might get failed if stuck there for long time ( I faced this issue once at lfortran )

**Comment by Planeshifter** [comment]
Okay, so there are currently problems with the AI scaffolding workflow when triggered from a fork, which we will need to look into.  Apologies!

As a temporary workaround, I have created a `assert/is-constantcase` branch with your README.md and the files: https://github.com/stdlib-js/stdlib/pull/541 

Feel free to either checkout that branch or copy the files over into this PR's branch.

**Comment by Pranavchiku** [comment]
I rebased it over my PR, and added implementation, made a clean build again, there are quite a few bugs like output not correctly predicted, linting errors, and a few things will be needed to fix before I can push those changes in this PR, will try to get that done as soon as possible :)

**Comment by kgryte** [comment]
Thanks for trying this out @Pranavchiku!

**Comment by Pranavchiku** [comment]
Sorry for delaying it, will try to get the things up as soon as possible.

**Comment by Pranavchiku** [comment]
so,
```javascript
tape( 'the command-line interface supports use as a standard stream', opts, function test( t ) {
	var cmd = [
		'printf "BEEP\nboop"',
		'|',
		EXEC_PATH,
		fpath
	];

	exec( cmd.join( ' ' ), done );

	function done( error, stdout, stderr ) {
		if ( error ) {
			t.fail( error.message );
		} else {
			t.strictEqual( stdout.toString(), 'true\ntrue\n', 'expected value' );
			t.strictEqual( stderr.toString(), '', 'does not print to `stderr`' );
		}
		t.end();
	}
});
```
This test is not passing in ```test.cli.js```, idk the reason, pasting the logs below
```
  the command-line interface supports use as a standard stream


    ✖ expected value
    -----------------
      operator: equal
      expected: 'false\nfalse\n'
      actual:   'false\n'
      at: done (/home/pranavchiku/stdlib/lib/node_modules/@stdlib/assert/is-constantcase/test/test.cli.js:179:6)

    ✔ does not print to `stderr`
    ```

File: lib/node_modules/@stdlib/assert/is-constantcase/bin/cli [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		}
+		str = data.toString();
+		if ( opts.sep ) {
+			str = str.split( opts.sep );
+			for ( i = 0; i < str.length; i++ ) {
+				console.log( isConstantCase( str[ i ] ) ); // eslint-disable-line no-console
+			}
+		} else {
+			console.log( isConstantCase( str ) ); // eslint-disable-line no-console
+		}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/bin/cli]: This logic is incorrect. See https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/assert/is-kebabcase/bin/cli#L80 for the correct logic.

cc @Planeshifter

**Comment by kgryte** [comment]
@Pranavchiku Left a comment. You may need to merge in changes from upstream. Currently, this PR is a bit "dirty" in that it contains changes/files which are already present in the upstream repo.

**Comment by Pranavchiku** [comment]
Yeah I will do that.

File: lib/node_modules/@stdlib/assert/is-constantcase/bin/cli (hunk 1 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/assert/is-constantcase/bin/cli]: oh okay, I actually will need to rewrite the tests.

Comment (kgryte) [reply lib/node_modules/@stdlib/assert/is-constantcase/bin/cli]: The tests should be fine, unless you changed them. It is just the logic in the CLI file which needs updating.

**Comment by Pranavchiku** [comment]
I observed few things on the files created by ```stdlib-bot```

1. Ending the file with two extra lines
2. Using extra line after `use-strict`
3. Tests logic are not great
4. ```test/fixtures``` file is not created

Rest everything was very perfect!

**Comment by kgryte** [comment]
@Pranavchiku That's great feedback! Thanks! cc @Planeshifter

File: lib/node_modules/@stdlib/assert/is-constantcase/bin/cli (hunk 1 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/assert/is-constantcase/bin/cli]: oh okay, I was not able to push those changes after I rebase it over my current PR, it threw me error, hence I had to change those.

**Comment by Pranavchiku** [comment]
I am currently getting a small error in ```test.cli.js```, I will resolve that, polish the PR and then push changes as soon as possible :)

**Comment by kgryte** [comment]
@Pranavchiku Awesome! Thanks for working on this!

**Comment by Pranavchiku** [comment]
Okay so in here ```the command-line interface supports specifying a custom delimiter when used as a standard stream (regexp)``` is not supported, I don't know the reason why it kept failing, if you can have a check

File: lib/node_modules/@stdlib/assert/is-constantcase/test/test.cli.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	function done( error, stdout, stderr ) {
+		if ( error ) {
+			t.fail( error.message );
+		} else {
+			t.strictEqual( stdout.toString(), 'false\ntrue\n', 'expected value' );
+			t.strictEqual( stderr.toString(), '', 'does not print to `stderr`' );
+		}
+		t.end();
+	}
+});
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/test/test.cli.js:206]: Missing tests for when the custom delimiter is a regexp. Can copy over from `@stdlib/assert/is-kebabcase` or similar package.

File: lib/node_modules/@stdlib/assert/is-constantcase/test/test.cli.js [hunk 3]
```diff
@@ -164,7 +164,7 @@ tape( 'the command-line interface tests if a string argument contains only upper
 
 tape( 'the command-line interface supports use as a standard stream', opts, function test( t ) {
 	var cmd = [
-		'printf "Beepboop"',
+		'printf "Beepboop\nbeepBoop"',
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/test/test.cli.js]: May be better to have both a positive and a negative test value here. E.g.,

```
Beepboop\nBEEPBOOP
```

File: lib/node_modules/@stdlib/assert/is-constantcase/test/test.cli.js [hunk 4]
```diff
@@ -205,6 +205,28 @@ tape( 'the command-line interface supports specifying a custom delimiter when us
 	}
 });
 
+tape( 'the command-line interface supports specifying a custom delimiter when used as a standard stream (regexp)', opts, function test( t ) {
+	var cmd = [
+		'printf \'beep boop\tcheckHere\'',
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/test/test.cli.js]: Same comment as above.

**Comment by Pranavchiku** [comment]
@kgryte you can have a final review.

File: lib/node_modules/@stdlib/assert/is-constantcase/README.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+```bash
+$ echo -n 'beep Boop' | is-constantcase
+false
+```
+
+By default, when used as a [standard stream][standard-streams], the implementation assumes newline-delimited data. To specify an alternative delimiter, set the `split` option.
+
+```bash
+$ echo -n 'beepBoop\tbeep_boop' | is-constantcase --split '\t'
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/README.md]:
```suggestion
$ echo -n 'beepBoop\tBEEP_BOOP' | is-constantcase --split '\t'
```

File: lib/node_modules/@stdlib/assert/is-constantcase/README.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+</section>
+
+<!-- /.notes -->
+
+<section class="examples">
+
+### Examples
+
+```bash
+$ is-constantcase beepboop
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/README.md]:
```suggestion
$ is-constantcase BEEP_BOOP
```

File: lib/node_modules/@stdlib/assert/is-constantcase/docs/repl.txt [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    Parameters
+    ----------
+    value: any
+        Value to test.
+
+    Returns
+    -------
+    bool: boolean
+        Boolean indicating whether value is a constantcase string.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/docs/repl.txt]:
```suggestion
        Boolean indicating whether a value is a constantcase string.
```

File: lib/node_modules/@stdlib/assert/is-constantcase/docs/types/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/**
+* Tests if a `value` is a constantcase `string`.
+*
+* @param value - value to test
+* @returns boolean indicating whether a `value` is a constantcase `string`
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/docs/types/index.d.ts]:
```suggestion
* Tests if a value is a constantcase string.
*
* @param value - value to test
* @returns boolean indicating whether a value is a constantcase string
```

File: lib/node_modules/@stdlib/assert/is-constantcase/lib/main.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+var constantcase = require( '@stdlib/string/base/constantcase' );
+
+
+// MAIN //
+
+/**
+* Tests if a `value` is a constantcase `string`.
+*
+* @param {*} value - value to test
+* @returns {boolean} boolean indicating whether a `value` is a constantcase `string`
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-constantcase/lib/main.js]:
```suggestion
* Tests if a value is a constantcase string.
*
* @param {*} value - value to test
* @returns {boolean} boolean indicating whether a value is a constantcase string
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Comment by Planeshifter** [comment]
> I observed few things on the files created by `stdlib-bot`
> 
> 1. Ending the file with two extra lines
> 2. Using extra line after `use-strict`
> 3. Tests logic are not great
> 4. `test/fixtures` file is not created
> 
> Rest everything was very perfect!

Thank you @Pranavchiku for the feedback. 1., 2., and 4. will be fixed in the next iteration of the scaffolding tool. We will keep iterating to hopefully get to a very smooth development experience in the near future.

**Pull request merged**